docs/config.md

```diff
@@ -0,0 +1,159 @@
+
+# File name to be used for the GPU metrics results
+[ filename_model_gpu: <string> | default: metrics-model-gpu.csv ]
+
+# File name to be used for storing the server only metrics.
+[ filename_server_only: <string> | default: metrics-server-only.csv ]
+
+# Specifies the maximum number of retries for any retry attempt.
+[ max_retries: <int> | default: 100 ]
+
+# Specifies how long (seconds) to gather server-only metrics
+[ duration_seconds: <int> | default: 5 ]
+
+# Duration of waiting time between each metric measurement in seconds
+[ monitoring_interval: <float> | default: 0.01 ]
+
+# The protocol used to communicate with the Triton Inference Server. Only 'http' and 'grpc' are allowed for the values.
+[ client_protocol: <string> | default: grpc ]
+
+# The full path to the perf_analyzer binary executable
+[ perf_analyzer_path: <string> | default: perf_analzyer ]
+
+# Time interval in milliseconds between perf_analyzer measurements.
+# perf_analyzer will take measurements over all the requests completed within
+# this time interval.
+[ perf_measurement_window: <int> | default: 5000 ]
+
+# Stops writing the output from the perf_analyzer to stdout.
+[ no_perf_output: <bool> | default: false ]
+
+# Triton Server version used when launching using Docker mode
+[ triton_version: <string> | default: 20.11-py3 ]
+
+# Logging level
+[ log_level: <string> | default: INFO ]
+
+# Triton Server HTTP endpoint url used by Model Analyzer client. Will be ignored if server-launch-mode is not 'remote'".
+[ triton_http_endpoint: <string> | default: localhost:8000 ]
+
+# Triton Server GRPC endpoint url used by Model Analyzer client. Will be ignored if server-launch-mode is not 'remote'".
+[ triton_grpc_endpoint: <string> | default: localhost:8001 ]
+
+# Triton Server metrics endpoint url used by Model Analyzer client. Will be ignored if server-launch-mode is not 'remote'".
+[ triton_metrics_url: <string> | default: localhost:8002 ]
+
+# The full path to the tritonserver binary executable
+[ triton_server_path: <string> | default: tritonserver ]
+
+# The full path to a file to write the Triton Server output log.
+[ triton_output_path: <string> ]
+
+# List of GPU UUIDs to be used for the profiling. Use 'all' to profile all the GPUs visible by CUDA."
+[ gpus: <string|comma-delimited-list-string> | default: 'all' ]
+```
+
+If you save configuration file in `config.yml`, you can provide
+the path to the Model Analyzer config using the 
+`-f` flag.
+
+```
+model-analyzer -f config.yml
+```
+
+All the flags supported in the CLI, are supported in the configuration file
+too.
+ ## Example
+
+An example configuration looks like below:
+
+```yaml
+model_repository: /path/to/model-repository
+triton_launch_mode: docker
+model_names:
+  - vgg_19_graphdef
+
+batch_sizes:
+    start: 4
+    end: 9
+concurrency:
+    - 2
+    - 4
+    - 8
+```
+
+If you save this file to the `config.yml`, you can start
+Model Analyzer using the `-f` flag.
```

Comment:
Is there a long name for the CLI? May want to specify that as well.

Reply:
Yes, we have a long name for the CLI too. It is `--config-file`. I will include this here too.